common/functions/src/scalars/dates/timeslot.rs

```diff
@@ -67,19 +47,29 @@ where T: NoArgTimeFunction + Clone + Sync + Send + 'static
         Ok(false)
     }
 
-    fn eval(&self, _columns: &DataColumnsWithField, input_rows: usize) -> Result<DataColumn> {
-        let value = T::execute();
-        Ok(DataColumn::Constant(
-            DataValue::UInt32(Some(value)),
-            input_rows,
-        ))
+    fn eval(&self, columns: &DataColumnsWithField, input_rows: usize) -> Result<DataColumn> {
+        let fmt = "%Y-%m-%d %H:%M:%S";
```

Comment:
Actually, we do not want this.

We must avoid Implicit conversion between string and DateTimes/Datetime.

The timeSlot function's prototype is `timeSlot( DateTime) ` or `timeSlot()` (which equals to `timeslot(now()` ).
You don't need to care about the format in timeSlot function.

We can have another function to cast string into DateTime (that's not an easy task, because the time formats are very complex to consider)